components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/APIClient.java

```diff
@@ -202,13 +221,24 @@ private void handleServerError(ActionInvocationResponse.Builder builder, HttpEnt
             builder.errorLog("Unexpected error response received for the status code: " + statusCode + ". "
                     + e.getMessage());
             builder.retry(true);
+            ActionInvocationErrorResponse invalidResponse = handleInvalidResponse();
+            builder.response(invalidResponse);
         }
     }
 
-    private ActionInvocationSuccessResponse handleSuccessResponse(HttpEntity responseEntity)
+    private ActionInvocationErrorResponse handleInvalidResponse() {
+
+        return new ActionInvocationErrorResponse.Builder()
+                .actionStatus(ActionInvocationResponse.Status.ERROR)
+                .error(SERVER_ERROR_CODE)
+                .errorDescription(SEVER_ERROR_MESSAGE)
+                .build();
+    }
+
+    private ActionInvocationResponse.APIResponse handleSuccessOrFailureResponse(HttpEntity responseEntity)
             throws ActionInvocationException {
 
-        return deserialize(responseEntity, ActionInvocationSuccessResponse.class);
```

Comment:
Looks like this method becomes obsolete with the new improvement. Is that the case ? If so remove that

Reply:
This is removed with 148073eab6305adc3e26c71ca755a0550e805f54